src/setuptools_scm/version.py

```diff
@@ -20,6 +20,38 @@
 SEMVER_LEN = 3
 
 
+class NonNormalizedVersion:
+    """A non-normalizing version handler.
+
+    You can use this class to disable all version verification and normalization.
+    For example you can use this to avoid git release candidate version tags
+    ("1.0.0-rc1") to be normalized to "1.0.0rc1". Only use this if you fully
+    trust the version tags.
+    """
+
+    def __init__(self, tag):
+        self.tag = tag
+
+    def __repr__(self):
+        return self.tag
+
+
+def _get_version_cls(config):
```

Comment:
this has to go

Reply:
Ok I move it to the constructor of `Configuration`. Note however that this imposes to import `Version` and define `NonNormalizedVersion` in the `config.py` package (less "purity")